packages/wrapped-keys-lit-actions/src/lib/common/batchGenerateEncryptedKeys.js

```diff
@@ -0,0 +1,147 @@
+const { encryptPrivateKey } = require('./internal/encryptKey');
+const {
+  generateEthereumPrivateKey,
+} = require('../ethereum/internal/generatePrivateKey');
+const { signMessageEthereumKey } = require('../ethereum/internal/signMessage');
+const {
+  generateSolanaPrivateKey,
+} = require('../solana/internal/generatePrivateKey');
+const { signMessageSolanaKey } = require('../solana/internal/signMessage');
+
+/* global accessControlConditions, actions, Lit*/
+
+async function processEthereumAction(action, ndx) {
+  const { network, generateKeyParams } = action;
+  const messageToSign = action.signMessageParams?.messageToSign;
+
+  const ethereumKey = generateEthereumPrivateKey();
+
+  const [generatedPrivateKey, messageSignature] = await Promise.all([
+    encryptPrivateKey({
+      accessControlConditions,
+      publicKey: ethereumKey.publicKey,
+      privateKey: ethereumKey.privateKey,
+    }),
+    messageToSign
+      ? signMessageEthereumKey({
+          messageToSign: messageToSign,
+          privateKey: ethereumKey.privateKey,
+        })
+      : Promise.resolve(),
+  ]);
+
+  return {
+    network,
+    generatedPrivateKey: {
+      ...generatedPrivateKey,
+      memo: generateKeyParams.memo,
```

Comment:
This is good to return since we're generating multiple keys

Reply:
:+1:
In theory the results coming back in the same _order_ that the array came in is 'enough', but being able to see the memo in the element you've 'zipped' together with an array you already had in your client context is pretty useful for debugging